whelk-core/src/main/groovy/whelk/Whelk.groovy

```diff
@@ -352,6 +353,13 @@ class Whelk {
             preUpdateDoc = doc.clone()
             updateAgent.update(doc)
             normalize(doc)
+
+            boolean detectCollisionsOnTypedIDs = false
+            List<Tuple2<String, String>> collidingIDs = getIdCollisions(doc, detectCollisionsOnTypedIDs)
+            if (!collidingIDs.isEmpty()) {
+                log.info("Refused update of " + doc.getShortId() + ". Document would have ID collision with : " + collidingIDs)
+                throw new WhelkRuntimeException("Update of ${doc.getShortId()} failed, due to ID collision with : " + collidingIDs)
+            }
```

Comment:
Maybe extract this to a method `checkIdCollsions(doc)` and use it in all three places?

Reply:
But the error messages are slighlty different, depending on if its CREATE or UPDATE.

Reply:
Do they have to be?
"`StorageCreateFailedException`" is not referenced in any other place.

Reply:
It's referenced here: rest/src/main/groovy/whelk/rest/api/Crud.groovy:729, apparently so we can return a http 409: Conflict.
I don't mean to say that's necessarily a meaningful thing to do, but I'd rather not change the crud interface if I don't have to.

Reply:
HAha, det behövs inte! PostgreSQL.verifyDocumentIdRetention() sköter redan det här. Förmodligen upkom den här i ett tidigare skede (den första stora inläsningen)!